### PR TITLE
Fix #2096 Maven archetype does not generate/include .gitignore file

### LINKED ISSUE #2096: Maven archetype does not generate/include .gitignore file
Include `.gitignore` in maven archetype

### PR DESCRIPTION
**Pull request by ppalaga**

### CODE REVIEW

File: devtools/common/src/main/resources/templates/gitignore.ftl [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+*.iws
+
+# NetBeans
+nb-configuration.xml
+
+# Visual Studio Code
+.vscode
+
+# OSX
+.DS_Store
```
Comment (gsmet) [inline devtools/common/src/main/resources/templates/gitignore.ftl]: If you can add the Vim temporary files
```
*.swp
*.swo
```
that would be nice.

And maybe the diff conflicts:
```
*.orig
*.rej
```
too.

**Review by gsmet — CHANGES_REQUESTED** [review]
Nice work. Thanks for taking care of it.

I just added a comment about useful additions to the `.gitignore` file, otherwise it looks good. Could you include them and squash everything? I'll merge then. Thanks!

File: devtools/common/src/main/java/io/quarkus/templates/rest/BasicRest.java [hunk 2]
```diff
@@ -154,6 +156,11 @@ private void createDockerIgnore() throws IOException {
         generate("templates/dockerignore.ftl", context, docker, "docker ignore");
     }
 
+    private void createGitIgnore() throws IOException {
+        File gitignore = new File(mkdirs(projectRoot), ".gitignore");
+        generate("templates/gitignore.ftl", context, gitignore, "docker ignore");
```
Comment (machi1990) [inline devtools/common/src/main/java/io/quarkus/templates/rest/BasicRest.java]: I think we have a typo: `resourceType` should be `"git ignore"` and not `"docker ignore"`.

**Comment by ppalaga** [comment]
Thanks for the review, in 6066907 : 

* `*.swp, *.swo, *.orig, *.rej` were added to gitignore.ftl and
* `"docker ignore"` was replaced with `"git ignore"`

**Review by gsmet — APPROVED** [review]

**Comment by gsmet** [comment]
Nice contribution, thanks!

**Pull request merged**